src/EPR.Payment.Facade/Services/Payments/OnlinePaymentsService.cs

```diff
@@ -116,62 +116,63 @@ private async Task<PaymentStatusResponseDto> GetPaymentStatusResponseAsync(strin
             }
         }
 
-        private UpdatePaymentRequestDto CreateUpdatePaymentRequest(PaymentDetailsDto paymentDetails, PaymentStatusResponseDto paymentStatusResponse, PaymentStatus status)
+        private UpdateOnlinePaymentRequestDto CreateOnlineUpdatePaymentRequest(OnlinePaymentDetailsDto onlinePaymentDetails, PaymentStatusResponseDto paymentStatusResponse, PaymentStatus status)
         {
-            var updateRequest = _mapper.Map<UpdatePaymentRequestDto>(paymentDetails);
-            updateRequest.GovPayPaymentId = paymentDetails.GovPayPaymentId;
+            var updateRequest = _mapper.Map<UpdateOnlinePaymentRequestDto>(onlinePaymentDetails);
+            updateRequest.GovPayPaymentId = onlinePaymentDetails.GovPayPaymentId;
             updateRequest.Status = status;
+            updateRequest.GovPayStatus = paymentStatusResponse.State?.Status;
             updateRequest.Reference = paymentStatusResponse.Reference;
             updateRequest.ErrorCode = paymentStatusResponse.State?.Code;
             updateRequest.ErrorMessage = paymentStatusResponse.State?.Message;
 
             return updateRequest;
         }
 
-        private async Task UpdatePaymentAsync(UpdatePaymentRequestDto updateRequest, Guid externalPaymentId, CancellationToken cancellationToken)
+        private async Task UpdateOnlinePaymentAsync(UpdateOnlinePaymentRequestDto updateRequest, Guid externalPaymentId, CancellationToken cancellationToken)
         {
             try
             {
-                await _httpPaymentsService.UpdatePaymentAsync(externalPaymentId, updateRequest, cancellationToken);
+                await _httpOnlinePaymentsService.UpdateOnlinePaymentAsync(externalPaymentId, updateRequest, cancellationToken);
             }
             catch (ValidationException ex)
             {
-                _logger.LogError(ex, LogMessages.ValidationErrorUpdatingPayment);
+                _logger.LogError(ex, LogMessages.ValidationErrorUpdatingOnlinePayment);
                 throw new ServiceException(ex.Message, ex);
             }
             catch (Exception ex)
             {
-                _logger.LogError(ex, LogMessages.UnexpectedErrorUpdatingPayment);
-                throw new ServiceException(ExceptionMessages.UnexpectedErrorUpdatingPayment, ex);
+                _logger.LogError(ex, LogMessages.UnexpectedErrorUpdatingOnlinePayment);
+                throw new ServiceException(ExceptionMessages.UnexpectedErrorUpdatingOnlinePayment, ex);
             }
         }
 
 
-        private static CompletePaymentResponseDto CreateCompletePaymentResponse(PaymentDetailsDto paymentDetails, PaymentStatusResponseDto paymentStatusResponse, PaymentStatus status)
+        private CompleteOnlinePaymentResponseDto CreateCompleteOnlinePaymentResponse(OnlinePaymentDetailsDto onlinePaymentDetails, PaymentStatusResponseDto paymentStatusResponse, PaymentStatus status)
```

Comment:
sonarqube: Member 'CreateCompleteOnlinePaymentResponse' does not access instance data and can be marked as static

Reply:
done